File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> indices = new HashMap<>();
        int[] res = new int[2];
        for(int i=0; i<nums.length; i++){
            if(indices.containsKey(target-nums[i])){
                res[0] = indices.get(target-nums[i]);
                res[1] = i;
                return res;
            }
            indices.put(nums[i], i);
        }
        return res;
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        seen = set()
        for i, num in enumerate(nums):
            if target-num in seen:
                return [i, map[target-num]]
            seen.add(num)
            map[num] = i
        return [-1,-1]
